### ISSUE TITLE
Mailu as mail relay Server

### ISSUE THREAD

**Issue by KarstenDE** [Mailu/Mailu#1594]
Hi community,

Possibly it is more a documentation then a technical issue ...

I want to run Mailu as a mail server at home. Based on the known issues this does not work for sending mail - based on spam protection rules at usual providers.

If I use a VPS cloud server as a mail relay server (with Mailu), how do I need to configure it? How to configure the SMTP authentication (at the cloud server)? Where to configure spam protection features like DKIM?

Thanks!

**Comment by Diman0** [comment]
The relevant options RELAYHOST and RELAYNETS can be found in the Mailu.env file documentation (https://mailu.io/1.7/configuration.html)


Homeserver:
If port 25 is blocked by your ISP, then possibly your ISP also offers SMPT servers you can use for sending email. You can use these servers to relay email for you. In the Netherlands almost all ISPs offer this service.
Regardless if you use your ISP's smtp server as smart host or Mailu on a VPS as a smart host, you need to configure in the mailu.env file the setting:
RELAYHOST=hostname:portnumber
RELAYUSER=login (usually an email address or email address prefix (part before @)
RELAYPASSWORD=password

RELAYUSER/RELAYPASSWORd is only required if the SMPT server requires authentication.
RELAYHOST instructs mailu to send all email via this relayhost (see the documentation I linked).
All docker containers must be restarted after adapting the mailu.env file to pick up the new settings.

VPS:
If mail relaying is not offered by your ISP, then you can enable email relaying on the VPS with the setting RELAYNETS= in the mailu.env file. This settings requires an IP Address. E.g.
RELAYNET=10.0.0.150 
Any IP address listed in RELAYNETS can send email via Mailu without authentication via port 25. Take care you enter the correct IP address 

In the SPF record for your domain name you must add the VPS MAilu server as allowed sender. E.g.
"v=spf1 mx a:domain_name_allowed sender ip4:11.111.11.111 -all"
The above sets the domain_name in  the MX record of your domain (@domain.com) as allowed sender, and the domain name domain_name and the IP address 11.111.11.111.
The DKIM and DMCARC records can be taken over as is (except for the p=reject, you want to use p=none for testing in the beginning)

For the VPS domain you can simply take over all the DNS settings from the admin gui of the mailu VPS server.

For testing I'd change the DMARC policy to not check the DKIM/SPF (p=none). When sending an email to gmail.com you can see in the email headers in your gmail account if SPF/DKIM and DMARC passed. After everything works you can change the policy to reject in your DMARC DNS record.

For more information on SPF/DKIM/DMARC see https://www.smartertools.com/blog/2019/04/09-understanding-spf-dkim-dmarc.
The DNS records of the domain in the Admin GUI can be reached via menu Mail Domains -> Details (first icon in action column). From this page you can also generate the DKIM/DMARC keys.

**Comment by Diman0** [comment]
You don't have to do anything. Mailu supports starttls out of box.



Op 20 aug. 2020 21:11, om 21:11, KarstenDE <notifications@github.com> schreef:

**Comment by KarstenDE** [comment]
Cool. With the Right Configure for the relayhost with port information it works.

`RELAYHOST=mail.domain.tld:587`

Port 25 is outbound blocked from my ISP and TLS enabled connection is better anyway.

Now my only issue is the same as at #1202: I can not send the mail via a Mailu relay server because the authentication user is not the Mail sender address :-(

I will close the issue here

**Comment by fracture-point** [comment]
I cannot for the life of me get this to work. I added this to mailu.env (version 2):

```
RELAYHOST=[email-smtp.us-east-2.amazonaws.com]:2565
RELAYUSER=iamuser
RELAYPASSWORD=iampass
```

I tested the creds and url separately using `swaks` and a test email is sent successfully (250 OK), however, when I use the above config with mailu I always get a `530` error...it is as though mailu doesn't even try to authenticate, even though I've configured a user and password:

```
smtp_1        | Feb 17 09:06:18 mail5 postfix/smtp[413]: 1438BBE25B0: to=<ping@tools.mxtoolbox.com>, relay=email-smtp.us-east-2.amazonaws.com[3.23.0.113]:587, delay=2, delays=1.2/0.16/0.65/0.03, dsn=5.0.0, status=bounced (host email-smtp.us-east-2.amazonaws.com[3.23.0.113] said: 530 Authentication required (in reply to MAIL FROM command))
```

I have also tried variations of the RELAYHOST config to no avail:
RELAYHOST=[email-smtp.us-east-2.amazonaws.com]:submission
RELAYHOST=email-smtp.us-east-2.amazonaws.com:587
RELAYHOST=email-smtp.us-east-2.amazonaws.com

I also cross-referenced the syntax in the Postfix docs (https://www.postfix.org/SASL_README.html Section "Enabling SASL authentication in the Postfix SMTP/LMTP client") with the formatting in the Mailu code (https://github.com/Mailu/Mailu/blob/9d0325fa1501c9e9febbbc7eaa5093cc6f8682c7/core/postfix/conf/sasl_passwd#L2) but I must be missing something here....

Any help appreciated!

**Comment by medram** [comment]
Just ensure to run this command after updating the mailu.env file to take effect:
```bash
docker-compose up -d
```

**Comment by mike-sirs** [comment]
> I cannot for the life of me get this to work. I added this to mailu.env (version 2):
> 
> ```
> RELAYHOST=[email-smtp.us-east-2.amazonaws.com]:2565
> RELAYUSER=iamuser
> RELAYPASSWORD=iampass
> ```
> 
> I tested the creds and url separately using `swaks` and a test email is sent successfully (250 OK), however, when I use the above config with mailu I always get a `530` error...it is as though mailu doesn't even try to authenticate, even though I've configured a user and password:
> 
> ```
> smtp_1        | Feb 17 09:06:18 mail5 postfix/smtp[413]: 1438BBE25B0: to=<ping@tools.mxtoolbox.com>, relay=email-smtp.us-east-2.amazonaws.com[3.23.0.113]:587, delay=2, delays=1.2/0.16/0.65/0.03, dsn=5.0.0, status=bounced (host email-smtp.us-east-2.amazonaws.com[3.23.0.113] said: 530 Authentication required (in reply to MAIL FROM command))
> ```
> 
> I have also tried variations of the RELAYHOST config to no avail: RELAYHOST=[email-smtp.us-east-2.amazonaws.com]:submission RELAYHOST=email-smtp.us-east-2.amazonaws.com:587 RELAYHOST=email-smtp.us-east-2.amazonaws.com
> 
> I also cross-referenced the syntax in the Postfix docs (https://www.postfix.org/SASL_README.html Section "Enabling SASL authentication in the Postfix SMTP/LMTP client") with the formatting in the Mailu code
> 
> https://github.com/Mailu/Mailu/blob/9d0325fa1501c9e9febbbc7eaa5093cc6f8682c7/core/postfix/conf/sasl_passwd#L2
> 
> 
> but I must be missing something here....
> Any help appreciated!

make sure you recreate smtp-1 container not just restart. it won't pull RELAY* params to postfix without it

**Comment by devsubngrill** [comment]
Hello The problem is you used in wrong format to use Aws SES relay the correct format is the following: [$SMTP_HOST]:587 $SMTP_USERNAME:$SMTP_PASSWORD

[Example.aws.ses]:587 username: password in the same line try this format is the correct

**Comment by fracture-point** [comment]
@medram I did indeed restart the docker services. @mike-sirs I know that the relay settings are being used because I'm receiving a 530 from AWS (if none of the relay settings were taking effect, Mailu wouldn't even know to try and send to AWS).

@devsubngrill I'm not following your suggestion - are you suggesting I concatenate the host,port,user,password all as the value of the RELAYHOST variable? e.g.

`RELAYHOST=[email-smtp.us-east-2.amazonaws.com]:587 iamuser:iampassword`

I did try this, regardless, and got the same error result (530 from AWS).
